### PR TITLE
tests: cleanup snapshot dir and reduce default guest memory

### PR DESCRIPTION
**Pull request by pupacha**
Improve test_snapshot_restore test to reduce transient issues.
* tests: Reduce Guest memory in _test_snapshot_restore
* tests: clean up snapshot dir after restore

### CODE REVIEW

File: tests/integration.rs [hunk 1]
```diff
@@ -6229,6 +6229,13 @@ mod common_parallel {
             &event_path_restored
         ));
 
+        // Remove the snapshot dir
+        Command::new("rm")
+            .arg("-rf")
+            .arg(snapshot_dir.as_str())
+            .output()
+            .unwrap();
+
```
Comment (rbradford) [inline tests/integration.rs:6238]: Please use this instead: https://doc.rust-lang.org/std/fs/fn.remove_dir_all.html

**Comment by pupacha** [comment]
Not really sure but looks like Quality Checks failures https://github.com/cloud-hypervisor/cloud-hypervisor/actions/runs/8878279980/job/24377497952?pr=6424 are caused from https://github.com/cloud-hypervisor/cloud-hypervisor/commit/e38a69f0d28f9c228d17229ded3cfbe51d0a8921

File: tests/integration.rs (hunk 1 above)
Comment (pupacha) [reply tests/integration.rs:6234]: Done.

**Comment by rbradford** [comment]
> Not really sure but looks like Quality Checks failures https://github.com/cloud-hypervisor/cloud-hypervisor/actions/runs/8878279980/job/24377497952?pr=6424 are caused from [e38a69f](https://github.com/cloud-hypervisor/cloud-hypervisor/commit/e38a69f0d28f9c228d17229ded3cfbe51d0a8921)

No they're from the beta toolchain gaining extra checks - @liuw is working on a PR - #6423

**Review by rbradford — APPROVED** [review]

**Pull request merged**